tests/integration.rs

```diff
@@ -374,13 +375,21 @@ mod subject_alt_names {
             params.alg = alg;
             let cert = rcgen::Certificate::from_params(params)
                 .expect("failed to make ee cert (this is a test bug)");
-            let der = cert
-                .serialize_der_with_signer(&issuer)
-                .expect("failed to serialize signed ee cert (this is a test bug)");
-            der
+            cert.serialize_der_with_signer(&issuer)
+                .expect("failed to serialize signed ee cert (this is a test bug)")
         };
 
-        expect_cert_dns_names(&ee_cert_derd, &[DNS_NAME]);
+        expect_cert_dns_names(&ee_cert_der, &[DNS_NAME]);
+    }
+
+    // This test reproduces https://github.com/rustls/webpki/issues/167 --- an
+    // end-entity cert where the common name is an empty SEQUENCE.
+    #[test]
+    pub fn empty_sequence_common_name() {
+        // handcrafted cert DER produced using `ascii2der`, since `rcgen` is
+        // unwilling to generate this particular weird cert.
+        let ee_cert_der = include_bytes!("misc/empty_sequence_common_name.der");
```

Comment:
For some of the other hand-crafted certs I've added recently that were produced with `ascii2der` I've also checked in the `.txt` representation that generated the DER, e.g.: [crl.idp.explicit.false.bool.der.txt](https://github.com/rustls/webpki/blob/main/tests/crls/crl.idp.explicit.false.bool.der.txt).

It isn't something we're doing today but eventually it would be nice to have CI check that the regenerating the `.der` from the `.txt` produces no diff. It also makes it easier for a reviewer to understand what the `.der` is doing when a new test case is added.

Reply:
I'm happy to check the text file in as well!